include/irr/asset/IMeshPacker.h

```diff
@@ -0,0 +1,123 @@
+#ifndef __IRR_I_MESH_PACKER_H_INCLUDED__
+#define __IRR_I_MESH_PACKER_H_INCLUDED__
+
+namespace irr
+{
+namespace asset
+{
+
+namespace meshPackerUtil
+{
+    struct AllocationParams
+    {
+        size_t indexBuffSupportedSizeInBytes                 = 2147483648ull;      /*  2GB*/
+        size_t vertexBuffSupportedSizeInBytes                = 2147483648ull;      /*  2GB*/
+        size_t perInstanceVertexBuffSupportedSizeInBytes     = 33554432ull;        /* 32MB*/
+        size_t MDIDataBuffSupportedSizeInBytes               = 2147483648ull;      /*  2GB*/
```

Comment:
If one MDI needs at least 64 triangles, and one triangle is 3 indices of 2 bytes each, and your index buffer can only be max 8GB (due to firstIndex being a `uint32_t` in the MDI struct), this gives you a max of 20'000'000 MDI structs

Reply:
I've mentioned that 2GB is too much here before

> If you have 2GB of uint16_t indices, then you have 341M triangles, if you split meshbuffers into batches of 32-1024 tris, then your max MDI count is 10.6M, given that each MDI struct is 20 bytes... works out to 214MB max

Reply:
STILL